// react-trafficlight/TrafficLight.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /* eslint-disable max-len */


var scale = 1 / 0.375;

var TrafficLight = function TrafficLight(_ref) {
  var onRedClick = _ref.onRedClick,
      onYellowClick = _ref.onYellowClick,
      onGreenClick = _ref.onGreenClick,
      RedOn = _ref.RedOn,
      YellowOn = _ref.YellowOn,
      GreenOn = _ref.GreenOn,
      Size = _ref.Size,
      BlackColor = _ref.BlackColor,
      DisabledColor = _ref.DisabledColor,
      RedColor = _ref.RedColor,
      YellowColor = _ref.YellowColor,
      GreenColor = _ref.GreenColor,
      props = _objectWithoutProperties(_ref, ['onRedClick', 'onYellowClick', 'onGreenClick', 'RedOn', 'YellowOn', 'GreenOn', 'Size', 'BlackColor', 'DisabledColor', 'RedColor', 'YellowColor', 'GreenColor']);

  return _react2.default.createElement(
    'svg',
    _extends({ width: Size + 'px', height: Size * scale + 'px', viewBox: '0 0 60 160', version: '1.1' }, props),
    _react2.default.createElement(
      'defs',
      null,
      _react2.default.createElement('circle', { style: { cursor: onRedClick ? 'pointer' : undefined }, id: 'redCirclePath', cx: '30', cy: '30', r: '20' }),
      _react2.default.createElement('circle', { style: { cursor: onYellowClick ? 'pointer' : undefined }, id: 'yellowCirclePath', cx: '30', cy: '80', r: '20' }),
      _react2.default.createElement('circle', { style: { cursor: onGreenClick ? 'pointer' : undefined }, id: 'greenCirclePath', cx: '30', cy: '130', r: '20' }),
      _react2.default.createElement(
        'filter',
        { x: '-50%', y: '-50%', width: '200%', height: '200%', filterUnits: 'objectBoundingBox', id: 'shadowFilter' },
        _react2.default.createElement('feGaussianBlur', { stdDeviation: '3', 'in': 'SourceAlpha', result: 'shadowBlurInner1' }),
        _react2.default.createElement('feOffset', { dx: '0', dy: '4', 'in': 'shadowBlurInner1', result: 'shadowOffsetInner1' }),
        _react2.default.createElement('feComposite', { 'in': 'shadowOffsetInner1', in2: 'SourceAlpha', operator: 'arithmetic', k2: '-1', k3: '1', result: 'shadowInnerInner1' }),
        _react2.default.createElement('feColorMatrix', { values: '0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0', type: 'matrix', 'in': 'shadowInnerInner1' })
      )
    ),
    _react2.default.createElement('rect', { fill: BlackColor, x: '0', y: '0', width: '60', height: '160', rx: '8' }),
    _react2.default.createElement('use', { fill: RedOn ? RedColor : DisabledColor, fillRule: 'evenodd', xlinkHref: '#redCirclePath' }),
    _react2.default.createElement('use', { fill: YellowOn ? YellowColor : DisabledColor, fillRule: 'evenodd', xlinkHref: '#yellowCirclePath' }),
    _react2.default.createElement('use', { fill: GreenOn ? GreenColor : DisabledColor, fillRule: 'evenodd', xlinkHref: '#greenCirclePath' }),
    _react2.default.createElement(
      'g',
      null,
      _react2.default.createElement('use', { onClick: onRedClick, fill: 'black', fillOpacity: '1', filter: 'url(#shadowFilter)', xlinkHref: '#redCirclePath' }),
      _react2.default.createElement('use', { onClick: onYellowClick, fill: 'black', fillOpacity: '1', filter: 'url(#shadowFilter)', xlinkHref: '#yellowCirclePath' }),
      _react2.default.createElement('use', { onClick: onGreenClick, fill: 'black', fillOpacity: '1', filter: 'url(#shadowFilter)', xlinkHref: '#greenCirclePath' })
    )
  );
};

TrafficLight.propTypes = {
  onRedClick: _propTypes2.default.func,
  onYellowClick: _propTypes2.default.func,
  onGreenClick: _propTypes2.default.func,
  RedOn: _propTypes2.default.bool,
  YellowOn: _propTypes2.default.bool,
  GreenOn: _propTypes2.default.bool,
  Size: _propTypes2.default.number,
  BlackColor: _propTypes2.default.string,
  DisabledColor: _propTypes2.default.string,
  RedColor: _propTypes2.default.string,
  YellowColor: _propTypes2.default.string,
  GreenColor: _propTypes2.default.string
};

TrafficLight.defaultProps = {
  onRedClick: undefined,
  onYellowClick: undefined,
  onGreenClick: undefined,
  RedOn: false,
  YellowOn: false,
  GreenOn: false,
  Size: 60,
  BlackColor: '#000000',
  DisabledColor: '#4A4A4A',
  RedColor: '#D0021B',
  YellowColor: '#F8E71C',
  GreenColor: '#7ED321'
};

exports.default = TrafficLight;
